src/drive/web/modules/views/Recent/index.jsx

```diff
@@ -0,0 +1,144 @@
+  const result = useQuery(query.definition, query.options)
+  const resultData = result.data || []
+
+  const dirIds = uniq(resultData.map(({ dir_id }) => dir_id))
+  const parentsQuery = buildParentsByIdsQuery(dirIds)
+  const parentsResult = useQuery(parentsQuery.definition, parentsQuery.options)
+
+  let parentsDocsById = {}
+
+  if (parentsResult.fetchStatus === 'loaded' && parentsResult.data.length > 0) {
+    parentsDocsById = keyBy(parentsResult.data, '_id')
+  }
+
+  return {
+    ...result,
+    data: resultData.map(file => ({
+      ...file,
+      displayedPath: get(parentsDocsById[file.dir_id], 'path')
+    }))
+  }
+}
+
+const RecentView = ({ router, children }) => {
+  const { isBigThumbnail, toggleThumbnailSize } = useContext(
+    ThumbnailSizeContext
+  )
+
+  const query = buildRecentQuery()
+  const result = useFilesQueryWithPath(query)
+
+  const isInError = result.fetchStatus === 'error'
+  const hasDataToShow = !isInError && (result.data && result.data.length > 0)
+  const isLoading = !hasDataToShow && result.fetchStatus === 'loading'
+  const isEmpty = !isLoading && !hasDataToShow
+
+  const navigateToFolder = useCallback(folderId => {
+    router.push(`/folder/${folderId}`)
+  })
+
+  const navigateToFile = useCallback(file => {
+    router.push(`/recent/file/${file.id}`)
+  })
+
+  return (
+    <Main>
+      <ModalStack />
+      <Topbar>
+        <Breadcrumb />
+        <Toolbar canUpload={false} canCreateFolder={false} disabled={false} />
+      </Topbar>
+
+      <SelectionBar />
+      <FileListv2>
+        <MobileFileListHeader
+          folderId={null}
+          canSort={false}
+          thumbnailSizeBig={isBigThumbnail}
+          toggleThumbnailSize={toggleThumbnailSize}
+        />
+        <FileListHeader
+          folderId={null}
+          canSort={false}
+          thumbnailSizeBig={isBigThumbnail}
+          toggleThumbnailSize={toggleThumbnailSize}
+        />
+        <FileListBodyV2 selectionModeActive={false}>
+          {isInError && <Oops />}
+          {isLoading && <FileListRowsPlaceholder />}
+          {isEmpty && <EmptyDrive canUpload={false} />}
+          {hasDataToShow && (
+            <div className={isMobileApp() ? 'u-ov-hidden' : ''}>
+              {result.data.map(file => (
+                <File
+                  key={file._id}
+                  attributes={file}
+                  displayedFolder={null}
+                  onFolderOpen={navigateToFolder}
+                  onFileOpen={navigateToFile}
+                  withSelectionCheckbox={true}
+                  withFilePath={true}
+                  withSharedBadge={true}
+                  isFlatDomain={true}
+                  thumbnailSizeBig={isBigThumbnail}
+                />
+              ))}
+              {result.hasMore && <LoadMore fetchMore={result.fetchMore} />}
+            </div>
+          )}
+        </FileListBodyV2>
+      </FileListv2>
+      {children}
+    </Main>
```

Comment:
I am a little bit concerned here that we repeat all the logic from the Drive view, I wonder if we could reuse the same exact views between the Recent and Drive views, only with query and options of the toolbar changed ?

Reply:
https://github.com/cozy/cozy-drive/pull/2041/files#r446060353

Reply:
Is this going to land in this PR ?

Reply:
Nop, already too much work done on this one. Coming with the Trash view

Reply:
OK thanks. Next time, I think it'd be helpful to have this kind of information in the description.